src/lib/support/DefaultStorageKeyAllocator.h

```diff
@@ -175,11 +175,11 @@ class DefaultStorageKeyAllocator
     static StorageKeyName BindingTable() { return StorageKeyName::FromConst("g/bt"); }
     static StorageKeyName BindingTableEntry(uint8_t index) { return StorageKeyName::Formatted("g/bt/%x", index); }
 
-    // Client Monitoring
+    // ICD Management
 
-    static StorageKeyName ClientMonitoringTableEntry(chip::FabricIndex fabric)
+    static StorageKeyName IcdManagementTableEntry(chip::FabricIndex fabric, uint16_t index)
     {
-        return StorageKeyName::Formatted("f/%x/cm", fabric);
+        return StorageKeyName::Formatted("f/%x/icd/%x", fabric, index);
```

Comment:
What cleans these things up on fabric removal?  Or is this handled elsewhere already?

Reply:
I think this could go in a subsequent PR. Spec also needs to be clarified / updated for the effect on the cluster when the fabric is removed.

Reply:
I have added a FabricTable delegate, and implemented the OnDelete method, calling a new RemoveAll() from the IcdManagement table.